
<launch>
  <!-- <include file="$(find com2009_simulations)/launch/nav_world.launch" /> -->
  <include file="$(find acs6121)/launch/arena.launch" />
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="slam_methods" default="gmapping" doc="slam type [gmapping, cartographer, hector, karto, frontier_exploration]"/>
  <arg name="configuration_basename" default="turtlebot3_lds_2d.lua"/>
  <arg name="open_rviz" default="true"/>
  <arg name="map_name" default="explore"/>

  <!-- SLAM: Gmapping, Cartographer, Hector, Karto, Frontier_exploration, RTAB-Map -->
  <include file="$(find turtlebot3_slam)/launch/turtlebot3_$(arg slam_methods).launch">
    <arg name="model" value="$(arg model)"/>
    <arg name="configuration_basename" value="$(arg configuration_basename)"/>
  </include>


  <!-- <arg name="filename" value="$(find real_robot)/maps/map2"/>
  <node name="map_save" pkg="map_server" type="map_saver" args="-f $(arg filename)" /> -->

  <!-- To be modified -->
  <arg name="map_file" default=" $(find real_robot)/maps/$(arg map_name).yaml"/>
  <arg name="initial_pose_x" default="-0.05"/>
  <arg name="initial_pose_y" default="-0.07"/>
  
  <!-- Other arguments -->
  <arg name="initial_pose_a" default="-3.12"/>
  <arg name="move_forward_only" default="false"/>

  <!-- Turtlebot3 Bringup -->
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- AMCL -->
  <include file="$(find turtlebot3_navigation)/launch/amcl.launch">
    <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
  </include>

  <!-- move_base -->
  <include file="$(find turtlebot3_navigation)/launch/move_base.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>

  <!-- rviz -->
  <node pkg="rviz" type="rviz" name="rviz" required="true"
    args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
  
  <!-- <node pkg="real_robot" type="moving.py" name="moving" output="screen"/> -->
</launch>